packages/react-native-sdk/src/providers/StreamCall.tsx

```diff
@@ -134,3 +151,161 @@ const ClearPushWSSubscriptions = () => {
+        if (lowPowerMode === null) {
+          StreamVideoReactNative.isLowPowerModeEnabled().then(
+            (isLowPowerMode: boolean) =>
+              handleLowPowerMode(isLowPowerMode, 'android')
+          );
+        }
+
+        if (!androidPowerSubscription) {
+          const sub = eventEmitter.addListener(
+            StreamVideoReactNative.POWER_MODE_EVENT,
+            (isLowPowerMode: boolean) =>
+              handleLowPowerMode(isLowPowerMode, 'android')
+          );
+          setAndroidPowerSubscription(sub);
+        }
+
+        if (thermalState === null) {
+          StreamVideoReactNative.startThermalStatusUpdates().then(
+            (initialState: string) =>
+              handleThermalState(initialState, 'android')
+          );
+        }
+
+        if (!androidThermalSubscription) {
+          const sub = eventEmitter.addListener(
+            StreamVideoReactNative.THERMAL_EVENT,
+            (status: string) => handleThermalState(status, 'android')
+          );
+          setAndroidThermalSubscription(sub);
+        }
+      } else if (Platform.OS === 'ios') {
+        if (lowPowerMode === null) {
+          DeviceState?.isLowPowerModeEnabled()?.then(
+            (isLowPowerMode: boolean) =>
+              handleLowPowerMode(isLowPowerMode, 'ios')
+          );
+        }
+
+        if (thermalState === null) {
+          DeviceState?.currentThermalState()?.then((state: any) =>
+            handleThermalState(state, 'ios')
+          );
+        }
+
+        if (!iosPowerSubscription) {
+          const sub = eventEmitter.addListener(
+            'isLowPowerModeEnabled',
+            (res: any) => {
+              handleLowPowerMode(res, 'ios');
+            }
+          );
+          setIosPowerSubscription(sub);
+        }
+
+        if (!iosThermalSubscription) {
+          const sub = eventEmitter.addListener(
+            'thermalStateDidChange',
+            (state: string) => {
+              handleThermalState(state, 'ios');
+            }
+          );
+          setIosThermalSubscription(sub);
+        }
+      }
+    } else {
+      if (Platform.OS === 'android') {
+        eventEmitter.removeAllListeners(
+          StreamVideoReactNative.POWER_MODE_EVENT
+        );
+        eventEmitter.removeAllListeners(StreamVideoReactNative.THERMAL_EVENT);
+        StreamVideoReactNative.stopThermalStatusUpdates();
+      } else if (Platform.OS === 'ios') {
+        eventEmitter.removeAllListeners('isLowPowerModeEnabled');
+        eventEmitter.removeAllListeners('thermalStateDidChange');
+      }
+    }
+  }, [
+    callingState,
+    lowPowerMode,
+    handleLowPowerMode,
+    thermalState,
+    handleThermalState,
+    eventEmitter,
+    StreamVideoReactNative,
+    StreamVideoReactNative.POWER_MODE_EVENT,
+    StreamVideoReactNative.THERMAL_EVENT,
+    DeviceState,
+    androidPowerSubscription,
+    androidThermalSubscription,
+    iosPowerSubscription,
+    iosThermalSubscription,
+  ]);
+
```

Comment:
wow a huge dependency array, for a simple two method implementation
please recheck how you can simplify this